### PR TITLE
Feature/user footprint api

### PR DESCRIPTION
**Pull request by liruchen32**
# Issue
[[會員制] 我的閱讀：造訪紀錄串接](https://app.asana.com/0/1203699264568808/1206379511127434/f)

# Notice

# Dependency
[go-api#665](https://github.com/twreporter/go-api/pull/665)

### CODE REVIEW

File: packages/core/src/constants/prop-types.js [hunk 1]
```diff
@@ -22,6 +22,7 @@ const articleMetaForBookmark = PropTypes.shape({
   desc: PropTypes.string.isRequired,
   thumbnail: PropTypes.string.isRequired,
   published_date: PropTypes.string,
+  post_id: PropTypes.string,
```
Comment (Aylie-Chou) [inline packages/core/src/constants/prop-types.js:25]: could we also add `category` prop type here XD

File: packages/redux/src/actions/__test__/mocks/footprints.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+          subcategory: {
+            id: '63206383207bf7c5f871622f',
+            key: '63206383207bf7c5f871622f',
+            latest_order: 0,
+            name: '美國',
+          },
+        },
+      ],
+      published_date: '2015-12-16T00:00:00Z',
+      is_external: false,
```
Comment (Aylie-Chou) [inline packages/redux/src/actions/__test__/mocks/footprints.js:77]: could we also add `bookmark_id` key in mock data

File: packages/redux/src/actions/footprints.js [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+/**
+ * get user's footprints
+ * @param {string} jwt - access_token granted for the user
+ * @param {number} userID -  id of user
+ * @param {number} offset - the offset of the request
+ * @param {number} limit - max amount of records per fetch
+ * @returns {Function} - function will be executed in Redux Thunk middleware
+ */
+export function getUserFootprints(jwt, userID, offset, limit) {
```
Comment (Aylie-Chou) [inline packages/redux/src/actions/footprints.js:69]: could we add default value for offset & limit
ex. limit = 10, offset = 0

File: packages/redux/src/reducers/__test__/mocks/footprints.js [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+                subcategory: {
+                  id: '63206383207bf7c5f871622f',
+                  key: '63206383207bf7c5f871622f',
+                  latest_order: 0,
+                  name: '美國',
+                },
+              },
+            ],
+            published_date: '2015-12-16T00:00:00Z',
+            is_external: false,
```
Comment (Aylie-Chou) [inline packages/redux/src/reducers/__test__/mocks/footprints.js:79]: same as this [comment](https://github.com/twreporter/twreporter-npm-packages/pull/498/files#r1481009851)

**Review by Aylie-Chou — CHANGES_REQUESTED** [review]
Great work to build up footprint actions & reducers 👍
Please have a look at comments for some details to check.

Also if the footprints state would handle `bookmark` related state, that state should also be updated when bookmark been created or deleted in other places.
Please also help to check this part 🙏

File: packages/core/src/constants/prop-types.js (hunk 1 above)
Comment (liruchen32) [reply packages/core/src/constants/prop-types.js:25]: fixed

File: packages/redux/src/actions/__test__/footprints.test.js [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+            payload: {
+              statusCode: 200,
+              data: mockGetUserFootprintsData,
+            },
+          }
+          expect(store.getActions()[0]).toEqual(requestExp)
+          expect(store.getActions()[1]).toEqual(successExp)
+        })
+    })
+    test('should dispatch an action created by `getUserFootprints` if updating failed', () => {
```
[the remark this answers is not in the document]
Comment (liruchen32) [reply packages/redux/src/actions/__test__/footprints.test.js:58]: fixed

File: packages/redux/src/actions/__test__/mocks/footprints.js (hunk 2 above)
Comment (liruchen32) [reply packages/redux/src/actions/__test__/mocks/footprints.js:77]: fixed

File: packages/redux/src/actions/footprints.js (hunk 3 above)
Comment (liruchen32) [reply packages/redux/src/actions/footprints.js:69]: fixed

File: packages/redux/src/reducers/__test__/mocks/footprints.js (hunk 4 above)
Comment (liruchen32) [reply packages/redux/src/reducers/__test__/mocks/footprints.js:79]: fixed

File: packages/redux/src/actions/__test__/mocks/footprints.js (hunk 2 above)
Comment (Aylie-Chou) [reply packages/redux/src/actions/__test__/mocks/footprints.js:77]: bookmark_id would be a string

**Review by Aylie-Chou — CHANGES_REQUESTED** [review]
as comment ~~

**Review by Aylie-Chou — APPROVED** [review]
LGTM

**Pull request merged**